src/App.tsx

```diff
@@ -1,34 +1,131 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TodoModal } from './components/TodoModal/TodoModal';
 import { Loader } from './components/Loader';
 
+const getFilterTodos = (
+  todos: Todo[],
+  statusFilter: string,
+  queryFilter: string,
+): Todo[] => {
+  let filteredTodos: Todo[] = [];
+
+  switch (statusFilter) {
+    case 'all': {
+      filteredTodos = todos;
+      break;
+    }
+
+    case 'active': {
+      filteredTodos = todos.filter(todo => todo.completed === false);
+
+      break;
+    }
+
+    case 'completed': {
+      filteredTodos = todos.filter(todo => todo.completed === true);
+
+      break;
+    }
+
+    default: filteredTodos = todos;
+  }
+
+  if (!queryFilter) {
+    return filteredTodos;
+  }
+
+  const queryLower = queryFilter.trim().toLowerCase();
+  const preparedTodos = filteredTodos.filter(({ title }) => {
+    const titleLower = title.toLowerCase();
+
+    return titleLower.includes(queryLower);
+  });
+
+  return preparedTodos;
+};
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState('all');
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
Better to set `false` as default value and change it to `true` only before you are making API call